src/main/java/org/opentripplanner/routing/edgetype/PathwayEdge.java

```diff
@@ -23,37 +24,55 @@
     public static final I18NString DEFAULT_NAME = new NonLocalizedString("pathway");
 
     private final I18NString name;
-    private int traversalTime;
-    private double distance;
-    private int steps;
-    private double angle;
-
-    private boolean wheelchairAccessible = true;
-    private FeedScopedId id;
-
-    public PathwayEdge(Vertex fromv, Vertex tov, I18NString name) {
-        super(fromv, tov);
-        this.name = Objects.requireNonNullElse(name, DEFAULT_NAME);
+    private final int traversalTime;
+    private final double distance;
+    private final int steps;
+    private final double angle;
+
+    private final boolean wheelchairAccessible;
+    private final FeedScopedId id;
+
+    public static PathwayEdge withComputedDistance(Vertex fromV, Vertex toV, I18NString name) {
+        return new PathwayEdge(
+                fromV,
+                toV,
+                null,
+                name,
+                0,
+                0, // distance is computed in the default constructor
+                0,
+                0,
+                true
+        );
     }
 
     public PathwayEdge(
-        Vertex fromv,
-        Vertex tov,
-        FeedScopedId id,
-        I18NString name,
-        int traversalTime,
-        double distance,
-        int steps,
-        double angle,
-        boolean wheelchairAccessible
+            Vertex fromv,
+            Vertex tov,
+            FeedScopedId id,
+            I18NString name,
+            int traversalTime,
+            double distance,
+            int steps,
+            double angle,
+            boolean wheelchairAccessible
     ) {
-        this(fromv, tov, name != null ? name : tov.getName());
+        super(fromv, tov);
+        this.name = Objects.requireNonNullElse(name, DEFAULT_NAME);
         this.id = id;
         this.traversalTime = traversalTime;
-        this.distance = distance;
         this.steps = steps;
         this.angle = angle;
         this.wheelchairAccessible = wheelchairAccessible;
+
+        // the GTFS spec allows you to define a pathway which has neither traversal time, distance
+        // nor steps. if that happens we compute it from the coordinates of the vertices
+        if (traversalTime <= 0 && distance <= 0 && steps <= 0) {
+            this.distance = distance(fromv.getCoordinate(), tov.getCoordinate());
+        }
+        else {
+            this.distance = distance;
+        }
```

Comment:
I find the placement of this somewhat confusing: there is a `withComputedDistance()`, but it doesn't actually compute the distance -- it happens in the constructor, which is always called. Which means that `withComputedDistance()` doesn't need to be used to have a computed distance.

Having a "normal" (non-elevator, non-boarding area) pathway with a 0 `length`/`traversal_time` will also have a computed distance.

The length could be calculated in `AddTransitModelEntitiesToGraph` when creating boarding areas and normal" pathways.